src/NUnitFramework/framework/TestContext.cs

```diff
@@ -476,16 +478,110 @@ public TestAdapter(Test test)
             /// <summary>
             /// The expected result if there is one for the test
             /// </summary>
-            public object? ExpectedResult
+            public object? ExpectedResult => (_test as TestMethod)?.ExpectedResult;
+
+            /// <summary>
+            /// The parent of this test or suite
+            /// </summary>
+            public ITest? Parent => _test.Parent;
+
+            /// <summary>
+            /// Returns all properties in the hierarchy
+            /// </summary>
+            /// <returns></returns>
+            public IDictionary<PropertyHierachyItem, IList> PropertyHierarchy()
+            {
+                var dict = new Dictionary<PropertyHierachyItem, IList>();
+                ITest? test = _test;
+                do
+                {
+                    foreach (var property in test.Properties.Keys)
+                    {
+                        var values = test.Properties[property];
+                        dict.Add(
+                            new PropertyHierachyItem { Name = property, Level = test.Name ?? string.Empty }, values);
+                    }
+                    test = test.Parent;
+                }
+                while (test is object);
+
+                return dict;
+            }
+
+            /// <summary>
+            /// Returns all values of a given property
+            /// </summary>
+            /// <param name="property">Name of property</param>
+            public IEnumerable<object> PropertyValues(string property)
+            {
+                var list = new List<object>();
+                var props = PropertyHierarchy();
+                foreach (var item in props.Keys.Where(o => o.Name == property))
+                {
+                    var values = props[item];
+                    foreach (var o in values)
+                        list.Add(o);
+                }
+
+                return list.Distinct();
```

Comment:
That makes sense to me too!

Reply:
Since the same properties can have multiple values, they will not override.  
If a fixture defines properties: A=1,2 and B=3,4
If a test defines properties: B=5.
Then the result would be  A= 1,2 and B=3,4,5.